crates/holochain_types/src/chain.rs

```diff
@@ -163,15 +163,29 @@ pub enum Sequences {
 /// Response to a `must_get_agent_activity` call.
 pub enum MustGetAgentActivityResponse {
     /// The activity was found.
-    Activity(Vec<RegisterAgentActivity>),
+    Activity {
+        /// The actions performed by the agent.
+        activity: Vec<RegisterAgentActivity>,
+        /// Any warrants issued to the agent for this activity.
+        warrants: Vec<WarrantOp>,
```

Comment:
How do I use this in a deterministic way? I think that's a question we ought to be able to answer before handing this to devs (or Holo).

I understand leaving ops in limbo once we know about warrants so they just never resolve. But I can't mark the op I'm validating as invalid because a warrant has shown up right? At least that's what I'd understood from the conversations about this, because other nodes might see the warrant at a different time.

Reply:
This is not the final return type of the must_get_agent_activity call, this is an intermediate value, and the warrants get peeled off before the actions are returned. Though your comment made me realize that when I peeled off the warrants, I wasn't doing anything with them, which is now no longer the case.